From d33dcefccfb3c6b362a936b45f04dfdd7e401bd6 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Fr=C3=A9d=C3=A9ric=20Pierret=20=28fepitre=29?=
 <frederic.pierret@qubes-os.org>
Date: Sat, 7 Nov 2020 16:35:55 +0100
Subject: [PATCH 1101/1103] docs: rename DATE to PANDOC_REL_DATE and allow to
 specify it

PANDOC_REL_DATE is now defined with respect to
SOURCE_DATE_EPOCH (if set). It improves reproducibility.
---
 INSTALL       | 1 +
 docs/Makefile | 4 ++--
 2 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/INSTALL b/INSTALL
index 2ce1f78b4d6a..84c93ab9d518 100644
--- a/INSTALL
+++ b/INSTALL
@@ -240,6 +240,7 @@ XEN_BUILD_DATE=<output of date(1)>
 XEN_BUILD_TIME=hh:mm:ss
 SMBIOS_REL_DATE=mm/dd/yyyy
 VGABIOS_REL_DATE="dd Mon yyyy"
+PANDOC_REL_DATE="+%Y-%m-%d"
 
 It is sufficient to define only SOURCE_DATE_EPOCH instead of the above
 variables for having reproducible builds. By default, they use
diff --git a/docs/Makefile b/docs/Makefile
index 8de1efb6f5bc..92ef1b5238f8 100644
--- a/docs/Makefile
+++ b/docs/Makefile
@@ -3,7 +3,7 @@ include $(XEN_ROOT)/Config.mk
 -include $(XEN_ROOT)/config/Docs.mk
 
 VERSION		:= $(shell $(MAKE) -C $(XEN_ROOT)/xen --no-print-directory xenversion)
-DATE		:= $(shell date +%Y-%m-%d)
+PANDOC_REL_DATE	?= $(shell date $(DATE_EPOCH_OPTS) +%Y-%m-%d)
 
 DOC_ARCHES      := arm x86_32 x86_64
 MAN_SECTIONS    := 1 5 7 8
@@ -98,7 +98,7 @@ man$(1)/%.$(1): man/%.$(1).pandoc Makefile
 ifneq ($(PANDOC),)
 	@$(INSTALL_DIR) $$(@D)
 	$(PANDOC) --standalone -V title=$$* -V section=$(1) \
-		      -V date="$(DATE)" -V footer="$(VERSION)" \
+		      -V date="$(PANDOC_REL_DATE)" -V footer="$(VERSION)" \
 			  -V header=Xen $$< -t man --output $$@
 else
 	@echo "pandoc not installed; skipping $$@"
-- 
2.37.3

